backend/user_service/controllers/userController.js

```diff
@@ -3,8 +3,10 @@ const admin = require('firebase-admin');
 
 // Function to create a new user and add them to the Firestore collection
 const addToUserCollection = async (req, res) => {
-    const { uid, email, displayName } = req.body;
-
+    const { uid, email, displayName, username } = req.body;
+    if (!/^[a-zA-Z0-9]+$/.test(username)) {
```

Comment:
Perhaps create a const for this regex and export it from somewhere. Would make your code clearer.

maybe in a file such as regex.ts

Reply:
Sounds good, i'll put it under /lib?